components/builder-api/src/server/handlers.rs

```diff
@@ -234,27 +235,56 @@ pub fn job_log(req: &mut Request) -> IronResult<Response> {
         .and_then(FromValue::from_value)
         .unwrap_or(false);
 
+    let mut job_get = JobGet::new();
     let mut request = JobLogGet::new();
     request.set_start(start);
     {
         let params = req.extensions.get::<Router>().unwrap();
         match params.find("id").unwrap().parse::<u64>() {
-            Ok(id) => request.set_id(id),
+            Ok(id) => {
+                request.set_id(id);
+                job_get.set_id(id);
+            }
             Err(e) => {
                 debug!("Error finding id. e = {:?}", e);
                 return Ok(Response::with(status::BadRequest));
             }
         }
     }
-    match route_message::<JobLogGet, JobLog>(req, &request) {
-        Ok(mut log) => {
-            if !include_color {
-                log.strip_ansi();
+
+    // Before fetching the logs, we need to check and see if the logs we want to fetch are for
+    // a job that's building a private package, and if so, do we have the right to see said
+    // package.
+    match route_message::<JobGet, Job>(req, &job_get) {
+        Ok(mut job) => {
+            let ident = job.take_package_ident();
+            let mut opg = OriginPackageGet::new();
+            opg.set_ident(ident);
+
+            if session_id.is_some() {
+                opg.set_account_id(session_id.unwrap());
+            }
+
+            match route_message::<OriginPackageGet, OriginPackage>(req, &opg) {
+                Ok(_) => {
+                    // We made it this far, which means either the package is public or it's
```

Comment:
I really like how this shook out. Really nice that getting past the PERMISSION NetError means you've got permissions ;)